### PR TITLE
Show actual error response from UID2 operator instead of generic error response from Java network library on SDK side

### PR DESCRIPTION
**Pull request by sunnywu**

### CODE REVIEW

File: src/main/java/com/uid2/client/PublisherUid2Client.java [hunk 1]
```diff
@@ -62,16 +61,28 @@ public TokenGenerateResponse generateTokenResponse(TokenGenerateInput tokenGener
...
             throw new Uid2Exception("Error communicating with api endpoint", e);
         }
     }
+    
+    private static String getResponse(Response response) {
+        try {
+            if(response == null) {
+                return "empty response";
+            }
+            return response.body() != null ? response.body().string() : response.toString();
```
Comment (sunnywu) [inline src/main/java/com/uid2/client/PublisherUid2Client.java:81]: i think it's better to return [response.toString(](url)) than "" here?

File: src/main/java/com/uid2/client/PublisherUid2Client.java [hunk 2]
```diff
@@ -62,16 +61,28 @@ public TokenGenerateResponse generateTokenResponse(TokenGenerateInput tokenGener
 
 
         try (Response response = client.newCall(request).execute()) {
+            final String responseString = getResponse(response);
+            
             if (!response.isSuccessful()) {
-                throw new Uid2Exception("Unexpected code " + response);
+                throw new Uid2Exception("Unexpected code " + responseString);
```
Comment (jon8787) [inline src/main/java/com/uid2/client/PublisherUid2Client.java:67]: could the above 2 lines could go in the method as well?

File: src/main/java/com/uid2/client/PublisherUid2Client.java (hunk 1 above)
Comment (sunnywu) [reply src/main/java/com/uid2/client/PublisherUid2Client.java:81]: it was the 

_Unexpected code Response{protocol=h2, code=400, message=, url=https://operator-integ.uidapi.com/v2/token/generate}_ 

like the APAC pub saw previously

File: src/main/java/com/uid2/client/PublisherUid2Client.java (hunk 2 above)
Comment (sunnywu) [reply src/main/java/com/uid2/client/PublisherUid2Client.java]: done

File: src/main/java/com/uid2/client/PublisherUid2Client.java [hunk 3]
```diff
@@ -62,16 +57,32 @@ public TokenGenerateResponse generateTokenResponse(TokenGenerateInput tokenGener
...
         }
     }
+    
+    private static String getResponse(Response response) {
+
+        String responseString = "";
+
+        try {
+            if(response == null) {
+                responseString = "empty response";
```
Comment (jon8787) [inline src/main/java/com/uid2/client/PublisherUid2Client.java:73]: when would that "empty response" string be consumed?

Comment (sunnywu) [reply src/main/java/com/uid2/client/PublisherUid2Client.java:73]: when `response ` is null which is never but just that to be defensive....

Comment (jon8787) [reply src/main/java/com/uid2/client/PublisherUid2Client.java:73]: Yeah I'm just wondering if we should be throwing instead

Comment (sunnywu) [reply src/main/java/com/uid2/client/PublisherUid2Client.java]: ok changed to throw

**Review by jon8787 — APPROVED** [review]

**Pull request merged**